test/process_bootstrap_test.cc

```diff
@@ -649,16 +649,100 @@ TEST_F(CreateBootstrapConfigurationTest, CreatesBootstrapForH2WithTls) {
   Envoy::MessageUtil::validate(*bootstrap, Envoy::ProtobufMessage::getStrictValidationVisitor());
 }
 
-TEST_F(CreateBootstrapConfigurationTest, FailsForUnimplementedH3) {
+TEST_F(CreateBootstrapConfigurationTest, CreatesBootstrapForH3) {
   uris_.push_back(std::make_unique<UriImpl>("https://www.example.org"));
   resolveAllUris();
 
   std::unique_ptr<Client::OptionsImpl> options =
       Client::TestUtility::createOptionsImpl("nighthawk_client --h3 https://www.example.org");
 
+  absl::StatusOr<Bootstrap> expected_bootstrap = parseBootstrapFromText(R"pb(
+    static_resources {
+      clusters {
+        name: "0"
+        type: STATIC
+        connect_timeout {
+          seconds: 30
+        }
+        circuit_breakers {
+          thresholds {
+            max_connections {
+              value: 100
+            }
+            max_pending_requests {
+              value: 1
+            }
+            max_requests {
+              value: 100
+            }
+            max_retries {
+            }
+          }
+        }
+        transport_socket {
+          name: "envoy.transport_sockets.quic"
+          typed_config {
+            [type.googleapis.com/envoy.extensions.transport_sockets.quic.v3.QuicUpstreamTransport] {
+              upstream_tls_context {
+                common_tls_context {
+                  alpn_protocols: "h3"
+                }
+                sni: "www.example.org"
+              }
+            }
+          }
+        }
+        load_assignment {
+          cluster_name: "0"
+          endpoints {
+            lb_endpoints {
+              endpoint {
+                address {
+                  socket_address {
+                    address: "127.0.0.1"
+                    port_value: 443
+                  }
+                }
+              }
+            }
+          }
+        }
+        typed_extension_protocol_options {
+          key: "envoy.extensions.upstreams.http.v3.HttpProtocolOptions"
+          value {
+            [type.googleapis.com/envoy.extensions.upstreams.http.v3.HttpProtocolOptions] {
+              common_http_protocol_options {
+                max_requests_per_connection {
+                  value: 4294937295
+                }
+              }
+              explicit_http_config {
+                http3_protocol_options {
+                  quic_protocol_options {
+                    max_concurrent_streams {
+                      value: 2147483647
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+    stats_flush_interval {
+      seconds: 5
+    }
+  )pb");
+  ASSERT_THAT(expected_bootstrap, StatusIs(absl::StatusCode::kOk));
+
   absl::StatusOr<Bootstrap> bootstrap =
       createBootstrapConfiguration(*options, uris_, request_source_uri_, number_of_workers_);
-  ASSERT_THAT(bootstrap, StatusIs(absl::StatusCode::kUnimplemented));
+  ASSERT_THAT(bootstrap, StatusIs(absl::StatusCode::kOk));
```

Comment:
nit: is there no ASSERT_OK in open source?

Reply:
AFAIK there isn't.